samples/boards/nrf52/mesh/onoff_level_app/src/mesh/device_composition.c

```diff
@@ -0,0 +1,340 @@
+{
+	int err = 0;
+	struct net_buf_simple *msg = model->pub->msg;
+	struct server_state *state = model->user_data;
+
+	state->current = net_buf_simple_pull_u8(buf);
+
+	if (state->previous == state->current) {
+		return;
+	}
+
+	if (state->model_instance == 0x01) {
+		light_state_current.OnOff = state->current;
+		update_light_state();
+
+	} else if (state->model_instance == 0x02) {
+		if (state->current == 0x01) {
+			nvs_light_state_save();
+		}
+	}
+
+	if (model->pub->addr != BT_MESH_ADDR_UNASSIGNED) {
+		bt_mesh_model_msg_init(msg,
+				       BT_MESH_MODEL_OP_2(0x82, 0x04));
+		net_buf_simple_add_u8(msg, state->current);
+
+		err = bt_mesh_model_publish(model);
+
+		if (err) {
+			printk("bt_mesh_model_publish err %d\n", err);
+		}
+	}
+
+	state->previous = state->current;
+}
+
+static void gen_onoff_set(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	gen_onoff_set_unack(model, ctx, buf);
+	gen_onoff_get(model, ctx, buf);
+}
+
+static void gen_onoff_cli_status(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	u8_t tmp8 = net_buf_simple_pull_u8(buf);
+	printk("Acknownledgement from GEN_ONOFF_SRV = %u\n", tmp8);
+}
+
+static void gen_level_get(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	struct net_buf_simple *msg = NET_BUF_SIMPLE(10);
+	struct server_state *state = model->user_data;
+
+	bt_mesh_model_msg_init(msg, BT_MESH_MODEL_OP_2(0x82, 0x08));
+
+	net_buf_simple_add_le16(msg, state->current);
+
+	if (bt_mesh_model_send(model, ctx, msg, NULL, NULL)) {
+		printk("Unable to send LEVEL_SRV Status response\n");
+	}
+}
+
+static void gen_level_set_unack(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	int err;
+	struct net_buf_simple *msg = model->pub->msg;
+	struct server_state *state = model->user_data;
+
+	state->current = net_buf_simple_pull_le16(buf);
+
+	if (state->previous == state->current) {
+		return;
+	}
+
+	if (state->model_instance == 0x01) {
+		light_state_current.power = state->current;
+		update_light_state();
+	}
+
+	if (model->pub->addr != BT_MESH_ADDR_UNASSIGNED) {
+
+		bt_mesh_model_msg_init(msg,
+				       BT_MESH_MODEL_OP_2(0x82, 0x08));
+
+		net_buf_simple_add_le16(msg, state->current);
+
+		err = bt_mesh_model_publish(model);
+
+		if (err) {
+			printk("bt_mesh_model_publish err %d\n", err);
+		}
+	}
+
+	state->previous = state->current;
+}
+
+static void gen_level_set(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	gen_level_set_unack(model, ctx, buf);
+	gen_level_get(model, ctx, buf);
+}
+
+static void gen_delta_set_unack(struct bt_mesh_model *model,
+			  struct bt_mesh_msg_ctx *ctx,
+			  struct net_buf_simple *buf)
+{
+	int err;
+	struct net_buf_simple *msg = model->pub->msg;
+	struct server_state *state = model->user_data;
+
```

Comment:
Redundant empty line. I'd prefer to have any initializations that happen from return values of function calls to be done separately from the variable declarations (concerning the two lines below).